src/main/java/ch/njol/skript/effects/EffPush.java

```diff
@@ -77,6 +77,10 @@ protected void execute(final Event e) {
 			final Vector mod = d.getDirection(en);
 			if (v != null)
 				mod.normalize().multiply(v.doubleValue());
+			if (!Double.isFinite(mod.getX()) || !Double.isFinite(mod.getY()) || !Double.isFinite(mod.getZ())) {
+				// Some component of the mod vector is not finite, so just stop
+				return;
+			}
```

Comment:
```suggestion
			try {
				mod.checkFinite();
			} catch (IllegalArgumentException exception) {
				return;
			}
```
Spigot has a method for checking

Reply:
Throwing and catching an exception is slower than just checking, as java has to create a stacktrace every time and that takes resources. The current way is better

Reply:
That's how Spigot implemented, but having a Spigot maintained method is better.

Reply:
As Moderocky said, handling an exception is a heavy task. Your suggestion is exactly what I did, and I got their feedback. I'm supporting the current way my pull handles the error.